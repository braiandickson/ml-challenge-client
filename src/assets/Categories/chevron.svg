<svg xmlns="http://www.w3.org/2000/svg" width="6" height="11" viewBox="0 0 6 11">
    <path fill="#000" fill-opacity=".45" fill-rule="evenodd" d="M610.995 2366.826L610.212 2366 605 2371.498 610.216 2377 611 2376.174 606.567 2371.498z" transform="matrix(-1 0 0 1 611 -2366)"/>
</svg>
