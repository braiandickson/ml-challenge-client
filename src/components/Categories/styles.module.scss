@import '../../styles/variables/colors';

.categories {
  max-width: 850px;
  margin-bottom: 20px;
}

.categorie {
  margin-right: 10px;

  &__chevron {
    fill: $dark-grey;
    margin-right: 10px;
  }

  &__name {
    color: $dark-grey;
    font-size: 14px;
  }
}
