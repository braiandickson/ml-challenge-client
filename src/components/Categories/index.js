import React from 'react';

import chevron from '../../assets/Categories/chevron.svg';

const Categories = ({ categories }) =>
  categories && (
    <div className="row full-width categories">
      {categories.map((categorie) => (
        <div key={categorie} className="row categorie">
          {categories[0] !== categorie && (
            <img
              className="categorie__chevron"
              src={chevron}
              alt="chevron icon"
            />
          )}
          <span className="categorie__name">{categorie}</span>
        </div>
      ))}
    </div>
  );

export default Categories;
