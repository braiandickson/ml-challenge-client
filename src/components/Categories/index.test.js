import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Categories from './index';

const data = ['ComputaciÃ³n', 'Mouses, Teclados y Controles', 'Mouses'];

describe('#Categories', () => {
  it('Mounts the Categories component correctly with data', () => {
    const component = mount(<Categories categories={data} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
