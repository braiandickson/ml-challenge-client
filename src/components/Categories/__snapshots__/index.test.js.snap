// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Categories Mounts the Categories component correctly with data 1`] = `
<Categories
  categories={
    Array [
      "Computación",
      "Mouses, Teclados y Controles",
      "Mouses",
    ]
  }
>
  <div
    className="row full-width categories"
  >
    <div
      className="row categorie"
      key="Computación"
    >
      <span
        className="categorie__name"
      >
        Computación
      </span>
    </div>
    <div
      className="row categorie"
      key="Mouses, Teclados y Controles"
    >
      <img
        alt="chevron icon"
        className="categorie__chevron"
        src="chevron.svg"
      />
      <span
        className="categorie__name"
      >
        Mouses, Teclados y Controles
      </span>
    </div>
    <div
      className="row categorie"
      key="Mouses"
    >
      <img
        alt="chevron icon"
        className="categorie__chevron"
        src="chevron.svg"
      />
      <span
        className="categorie__name"
      >
        Mouses
      </span>
    </div>
  </div>
</Categories>
`;
